app/uk/gov/hmrc/gform/validation/DateValidation.scala

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.validation
+import java.time.LocalDate
+import uk.gov.hmrc.gform.validation.DateValidationLogic._
+
+import cats.{ Monoid, Semigroup }
+import cats.data.Validated._
+import cats.data._
+import cats.implicits._
+import uk.gov.hmrc.gform.sharedmodel.form.FormDataRecalculated
+import uk.gov.hmrc.gform.sharedmodel.formtemplate._
+import uk.gov.hmrc.gform.typeclasses.Now
+import uk.gov.hmrc.gform.validation.ComponentsValidator.errors
+import uk.gov.hmrc.gform.validation.ValidationServiceHelper._
+import uk.gov.hmrc.gform.validation.ValidationUtil._
+import uk.gov.hmrc.gform.validation.ComponentsValidator._
+import uk.gov.hmrc.gform.views.html.localisation
+import uk.gov.hmrc.gform.validation.DateValidation._
+
+import scala.util.{ Failure, Success, Try }
+
+case class DateValidation(data: FormDataRecalculated) {
```

Comment:
All method on this class should be marked as `private` if they can be. And any method which takes `data: FormDataRecalculated` as a parameter should reuse `data` from constructor.